app/src/main/java/com/sopt/now/ui/login/LoginActivity.kt

```diff
@@ -71,29 +47,59 @@ class LoginActivity : BindingActivity<ActivityLoginBinding>(R.layout.activity_lo
         }
     }
 
-    private fun validateLogin(): Boolean {
-        return validateId() && validatePassword()
+    private fun initButton() {
+        binding.btnLoginToSignup.setOnClickListener {
+            initSignUpBtnClickListener()
+        }
+
+        binding.btnLogin.setOnClickListener {
+            loginButtonEvent()
+        }
+    }
+
+    private fun initObserver() {
+        viewModel.liveData.observe(this) { state ->
+            when (state) {
+                is UiState.Loading -> {
+
+                }
+
+                is UiState.Success -> {
+                    navToHome(state.data)
+                }
+
+                is UiState.Error -> {
+                    toast(state.errorMessage)
+                }
+            }
+        }
     }
 
-    private fun validateId(): Boolean {
-        require(binding.etLoginId.text.toString() == user.id) {
-            toast(TOAST_NOT_EQUAL_ID)
-            return false
+    private fun navToHome(user: User) {
+        val intent = Intent(this, MainActivity::class.java)
+        with(intent) {
+            putExtra(TAG_USER, user)
+            startActivity(this)
```

Comment:
```suggestion
        Intent(this, MainActivity::class.java).apply {
             putExtra(TAG_USER, user)
        }
        startActivity(intent)
```

scope함수 사용할때 차이점을 알고 사용하는게 좋을 것 같습니다!

startActivity는 Intent 객체를 사용하는 Context의 함수입니다. 따라서 startActivity(this)를 호출하려면 this가 Context를 가리키도록 해야 합니다. 하지만 with 스코프 내에서 this는 Intent 객체를 참조합니다. 따라서 startActivity(this) 사용이 올바르지 않을 것 같습니다!

Reply:
음 이렇게 보니까 한줄짜리라 굳이 scope안써도 될것 같네요!

Reply:
scope 함수 아직 너무 어려운거 같아요.. 
this가 Context를 가리키도록 해야하는지 몰랐어요.. 감사합니다 어렵네요